Dockerfile

```diff
@@ -29,6 +29,8 @@ ENV _CUDA_COMPAT_TIMEOUT=90
 
 # Install other dependencies
 RUN pip install "h5py>=3.7.0" "mpi4py>=3.1.4" "netcdf4>=1.6.3" "ruamel.yaml>=0.17.22" "scikit-learn>=1.0.2" 
+RUN pip install "hydra-core>=1.2.0" "termcolor>=2.1.1" "wandb>=0.13.7" "mlflow>=2.1.1" "pydantic>=1.10.2" "imageio>=2.28.1" "moviepy>=1.0.3" "tqdm>=4.60.0"
```

Comment:
Why here instead of in `pyproject.toml`?

Reply:
In the builder stage, we want to only install the dependencies and not the source (because of the CI image referencing the builder). Unfortunately, this means that this install will be repeated in toml and dockerfile. Ideally, pip should support a `--only-deps` option, but that is still being considered (Ref: https://www.github.com/pypa/pip/issues/11440). The alternatives AFAIK are writing a separate requirements.txt either ourselves or through some pip-tools. I don't think it is a very good alternative, so I have been wanting to revisit this once the PyPi team implemented that feature. That being said, I feel like at the cost of having to maintain dependencies at two places, it is still worth doing it this way, because shipping docker image places a lot of constraints on what software can go in due to license issues etc. Consider tensorly-torch for example, that package's latest wheel is not compliant with our container requirements. We have a patch pushed, but the wheel is not public yet, so we have to manually install it inside our docker container following the process currently implemented.

Reply:
The `--only-deps` option is nice. I agree, can wait for that to remove this duplication. But maybe we should add a test for a source only install in a vanilla cuda container in the interim?

Reply:
Yes certainly! I think we should make this part of the QA - i.e. test the pip/source installs in a conda env or in a vanilla cuda container as you suggest. Thanks @akshaysubr, marking this as resolved now.